tests/MongoDB.Driver.Tests/Jira/CSharp3685Tests.cs

```diff
@@ -0,0 +1,47 @@
+﻿/* Copyright 2010-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization;
+using MongoDB.Bson.Serialization.Serializers;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.Core.Events;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Jira
+{
+    public class CSharp3685Tests
+    {
+        [Fact]
+        public void Registered_GuidSerializer_should_affect_resulting_message_serialization()
+        {
+            BsonSerializer.RegisterSerializer(new GuidSerializer(GuidRepresentation.Standard));
+
+            var events = new EventCapturer().Capture<CommandStartedEvent>(x => x.CommandName == "find");
+            var client = DriverTestConfiguration.CreateDisposableClient(events);
+            var database = client.GetDatabase("d");
+            var collection = database.GetCollection<BsonDocument>("c");
+
+            var guid = new Guid("4646d20b-7aa4-4ae0-9d4d-5ef417676b1c");
+            collection.FindSync(new BsonDocument("_id", new BsonBinaryData(guid, GuidRepresentation.Standard)));
```

Comment:
The test skips first serialization. It shows that even if the filter already had the desired GuidRepresentation it is still overwritten on the binary serialization step.